feat: Update FlexSearch and Add Support for All Languages

pourmand1376 opened the pull request:
This PR accomplishes two tasks:

- Updates FlexSearch to latest version (`npm install flexsearch@0.8.205`)
- Enhances encoding to support all languages by default.

Previous configs didn't support Farsi and Arabic by default. It assume only english text.

pourmand1376 committed:
chore(deps): update flexsearch to version 0.8.205 and adjust search encoder.

pourmand1376 committed:
refactor(search): enhance search encoder and update search results type

- Improved the encoder function to filter out empty tokens.
- Updated the search results type from a specific FlexSearch type to a more generic 'any' type for flexibility.
- Removed redundant rtl property from the index configuration.

pourmand1376 committed:
refactor(search): remove rtl property from search index configuration

pourmand1376 committed:
refactor(search): improve encoder function formatting

- Updated the encoder function to use consistent arrow function syntax for better readability.

jackyzha0 commented on quartz/components/scripts/search.inline.ts:
```diff
@@ -397,7 +403,7 @@ async function setupSearch(searchElement: Element, currentSlug: FullSlug, data:
     searchType = currentSearchTerm.startsWith("#") ? "tags" : "basic"
 
-    let searchResults: FlexSearch.SimpleDocumentSearchResultSetUnit[]
+    let searchResults: any[]
```
avoid typing with any

pourmand1376 replied:
Done.

pourmand1376 committed:
refactor(search): update search results type to DefaultDocumentSearchResults

- Imported DefaultDocumentSearchResults from FlexSearch for improved type safety.
- Changed the type of searchResults from 'any' to DefaultDocumentSearchResults<Item> for better clarity and maintainability.

jackyzha0 reviewed (approved):
looks good, can you double check with some CJK as well? (chinese japanese and korean characters)

pourmand1376 commented:
Yep. It works as well (I have tested Farsi extensively before). 

I tested some Japanese character from [here](https://www.symbolcopy.com/japanese-symbols.html) and it seems to work perfectly well.
